src/jvm/test/org/jetbrains/kotlinx/lincheck_test/gpmc/PrintingTest.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.gpmc
+
+import org.jetbrains.kotlinx.lincheck.ExperimentalModelCheckingAPI
+import org.jetbrains.kotlinx.lincheck.runConcurrentTest
+import org.junit.Ignore
+import org.junit.Test
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.CyclicBarrier
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+
+@OptIn(ExperimentalModelCheckingAPI::class)
+class PrintingTest {
+
+    @Ignore("Println's hang the execution")
+    @Test(timeout = TIMEOUT)
+    fun testPrintsWithLatch() {
+        runConcurrentTest(1) {
+            val numberOfThreads = 2
+            val barrier = CyclicBarrier(numberOfThreads) {
```

Comment:
I meant something even simpler than `CyclicBarrier` : )
I.e., just atomic integer or something.

Otherwise there is always a possibility that the test fails/hungs because of the primitive used, not because of `println`.